torch/testing/_internal/common_methods_invocations.py

```diff
@@ -10496,6 +10537,8 @@ def error_inputs_mean(op_info, device, **kwargs):
                        # RuntimeError: "nonzero_cuda" not implemented for 'ComplexHalf'
                        DecorateInfo(unittest.expectedFailure, 'TestSparseCSR', 'test_sparse_csr_consistency',
                                     dtypes=(torch.chalf,)),
+                       DecorateInfo(unittest.expectedFailure, 'TestSparseCompressed', 'test_consistency',
```

Comment:
It seems almost easier to add ComplexHalf support to nonzero than to add this DecorateInfo to all these tests. From a brief look at it, perhaps it's enough to just [extend the types that we're generating code for](https://github.com/pytorch/pytorch/blob/35d4a805ebc3b6eca1bafb2d332dffa8d0c1fc54/aten/src/ATen/native/cuda/Nonzero.cu#L121-L124)?

Reply:
There's a `AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND4` and it seems to be used for pretty much exactly this case: https://github.com/pytorch/pytorch/blob/f7ee308dfb6c1f28421562eca7685a7043c9f845/aten/src/ATen/native/cpu/CopyKernel.cpp#L84-L86

I'd still send that change in a separate PR in case it touches some unexpected piece, but adding `kComplexHalf` to nonzero seems worthwhile based on how many tests you need to skip here and given that it's needed for construction of COO Tensors in general.

Reply:
Resolved via https://github.com/pytorch/pytorch/pull/79062